### PR TITLE
Support profiling feature when building for UWP.

### PR DESCRIPTION
**Pull request by jdm**

### CODE REVIEW

**Comment by jdm** [comment]
The TravisCI failures will be fixed by #220.

File: mozjs/js/src/builtin/Profilers.cpp [hunk 1]
```diff
@@ -24,8 +24,13 @@
 #endif
 
 #ifdef XP_WIN
-#  include <process.h>
-#  define getpid _getpid
+#  ifndef JS_ENABLE_UWP
```
Comment (asajeffrey) [inline mozjs/js/src/builtin/Profilers.cpp]: Nit: can we make this an `ifdef` rather than an `ifndef`?

File: mozjs/js/src/builtin/Profilers.cpp [hunk 2]
```diff
@@ -24,8 +24,13 @@
...
+#    include <process.h>
+#    define getpid _getpid
+#else
```
Comment (asajeffrey) [inline mozjs/js/src/builtin/Profilers.cpp]: Nit: indentation.

**Comment by nox** [comment]
Why was it needed to disable vtune?

File: mozjs/js/moz.configure [hunk 3]
```diff
@@ -271,13 +271,13 @@ def profiling(value):
...
-@depends(profiling, target)
-def imply_vtune(value, target):
+@depends(profiling, target, js_enable_uwp)
+def imply_vtune(value, target, uwp):
     ok_cpu    = target.cpu in ['x86', 'x86_64']
     ok_kernel = target.kernel == 'WINNT' or \
                 (target.kernel == 'Linux' and target.os == 'GNU')
 
-    if value and ok_cpu and ok_kernel:
+    if value and ok_cpu and ok_kernel and not uwp:
```
Comment (nox) [inline mozjs/js/moz.configure]: Why is this patch needed if we explicitly disable `vtune` when invoking `configure`? I don't get it.

Comment (jdm) [reply mozjs/js/moz.configure]: Without being patched, configute complained that --enable-profiling always enabled vtune and you could not disable it.

Comment (nox) [reply mozjs/js/moz.configure]: Oh so `--disable-vtune` has less weight than this implied setting? Good to know.

**Pull request merged**